mrbgems/mruby-pack/src/pack.c

```diff
@@ -258,22 +261,22 @@ unpack_l(mrb_state *mrb, const unsigned char *src, int srclen, mrb_value ary, un
   if (flags & PACK_FLAG_SIGNED) {
     int32_t sl = ul;
 #ifndef MRB_INT64
-    if (!FIXABLE(sl)) {
+    if (INT_OVERFLOW_P(sl)) {
```

Comment:
I admit `FIXABLE` is no longer a sufficient check. In the past, `fixnum` integers need to be fit in 31bits under word-boxing mode. But I fixed it recently.

But I have noticed that we are in `ifndef MRB_INT64` (`mrb_int` is `int32_t`) and `sl` is `int32_t` so that `INT_OVERFLOW_P` is always false. Probably we have to check if `sl` is negative (too big unsigned integer converted to signed) or call `UINT_OVERFLOW_P()` with `ul`.

Reply:
Thank you for pointing out.
When I think about it again, confirmation is not necessary for `int32_t` (Because it is processing unsigned integers to signed integers for convenience of processing).

In summary:

|                |  `MRB_INT32`  |  `MRB_INT64`  |
| -------------: | ------------: | ------------: |
|   signed int32 | no-need check | no-need check |
| unsigned int32 |    need check | no-need check |
|   signed int64 |    need check | no-need check |
| unsigned int64 |    need check |    need check |

I'm sorry, I'll fix it tomorrow (maybe)...